cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,444 @@
+		return ServiceDeploymentResult{}, fmt.Errorf("failed merging kube configs: %w", err)
+	}
+
+	if _, err := t.kubectl.ConfigUseContext(ctx, clusterName, nil); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed using kube context '%s', %w", clusterName, err)
+	}
+
+	kubeFlags := kubectl.KubeCliFlags{
+		Namespace: namespace,
+		DryRun:    "client",
+		Output:    "yaml",
+	}
+
+	progress <- "Creating k8s namespace"
+	namespaceResult, err := t.kubectl.CreateNamespace(
+		ctx,
+		namespace,
+		&kubectl.KubeCliFlags{DryRun: "client", Output: "yaml"},
+	)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed creating kube namespace: %w", err)
+	}
+
+	_, err = t.kubectl.ApplyPipe(ctx, namespaceResult.Stdout, nil)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube namespace: %w", err)
+	}
+
+	progress <- "Creating k8s secrets"
+	secretResult, err := t.kubectl.CreateSecretGenericFromLiterals(ctx, "azd", t.env.Environ(), &kubeFlags)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed setting kube secrets: %w", err)
+	}
+
+	_, err = t.kubectl.ApplyPipe(ctx, secretResult.Stdout, nil)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube secrets: %w", err)
+	}
+
+	log.Printf("logging into registry %s\n", loginServer)
+
+	progress <- "Logging into container registry"
+	if err := t.az.LoginAcr(ctx, t.docker, t.env.GetSubscriptionId(), loginServer); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed logging into registry '%s': %w", loginServer, err)
+	}
+
+	imageTag, err := t.generateImageTag()
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed generating image tag: %w", err)
+	}
+
+	fullTag := fmt.Sprintf(
+		"%s/%s",
+		loginServer,
+		imageTag,
+	)
+
+	// Tag image.
+	log.Printf("tagging image %s as %s", path, fullTag)
+	progress <- "Tagging image"
+	if err := t.docker.Tag(ctx, t.config.Path(), path, fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed tagging image: %w", err)
+	}
+
+	log.Printf("pushing %s to registry", fullTag)
+
+	// Push image.
+	progress <- "Pushing container image"
+	if err := t.docker.Push(ctx, t.config.Path(), fullTag); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed pushing image: %w", err)
+	}
+
+	// Save the name of the image we pushed into the environment with a well known key.
+	t.env.SetServiceProperty(t.config.Name, "IMAGE_NAME", fullTag)
+
+	if err := t.env.Save(); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("saving image name to environment: %w", err)
+	}
+
+	progress <- "Applying k8s manifests"
+	t.kubectl.SetEnv(t.env.Values)
+	deploymentPath := t.config.K8s.DeploymentPath
+	if deploymentPath == "" {
+		deploymentPath = defaultDeploymentPath
+	}
+
+	err = t.kubectl.ApplyFiles(
+		ctx,
+		filepath.Join(t.config.RelativePath, deploymentPath),
+		&kubectl.KubeCliFlags{Namespace: namespace},
+	)
```

Comment:
should the namespace be a required parameter for `ApplyFiles` ?  (instead of using the entire KubeCliFlags?
Or does it work if we pass `nil` for the flags?

Reply:
From a pure kubectl cli perspective namespaces are not required and would like to keep our CLI matching this even though we would typically recommend users create resources within a namespace.